managed/services/supervisord/pmm_config.go

```diff
@@ -0,0 +1,219 @@
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package supervisord provides facilities for working with Supervisord.
+package supervisord
+
+import (
+	"bytes"
+	"os"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+// SavePMMConfig renders and saves pmm config.
+func SavePMMConfig(disableDB bool) error {
+	cfg, err := marshalConfig(disableDB)
+	if err != nil {
+		return err
+	}
+	if err := saveConfig(pmmConfig, cfg); err != nil {
+		return errors.Wrapf(err, "failed to save pmm config")
+	}
+	return nil
+}
+
+func marshalConfig(disableDB bool) ([]byte, error) {
+	var buf bytes.Buffer
+	if err := pmmTemplate.Execute(&buf, map[string]bool{"DisableInternalDB": disableDB}); err != nil {
+		return nil, errors.Wrapf(err, "failed to render pmm template")
+	}
+	return buf.Bytes(), nil
+}
```

Comment:
Having a boolean input param is a code smell that usually severely hinders code readability.
What's the difference between `supervisord.SavePMMConfig(false)` and `supervisord.SavePMMConfig(true)`?
You don't know unless you know the implementation of that function.

Often, the need for a bool param means that the function does more than it should and should therefore be split into two (or more) functions.

In this case, I don't think that the function should be split but I would change the input param from a `bool` to a `map[string]any` that will ultimately be the input of `Template.Execute`.
I see 2 advantages of this approach:
1. Extract the template configuration to the caller, increasing readability `supervisord.SavePMMConfig(map[string]bool{"DisableInternalDB": disableDB})` 
2. The template becomes extensible allowing for future customization of other fields other that `DisableInternalDB`

What do you think?